sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ResourceId.java

```diff
@@ -508,6 +507,29 @@ public boolean equals(ResourceId other) {
         return Arrays.equals(this.getValue(), other.getValue());
     }
 
+    @Override
+    public boolean equals(Object object) {
```

Comment:
This change is a bit concerning to me. I will verify if equals is being called anywhere for ResourceId. 
@moderakh  - do you know any of the places where we do so ?